### PR TITLE
Added a dropdown button component.

### PR DESCRIPTION
**Pull request by AshleyKensington**
Created Dropdown button 
Added ...rest to the icon on the button so that it inherits the props from the button

### CODE REVIEW

File: src/components/DropdownButton/DropdownButton.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  renderButtonText = title => {
+    return (
+      <React.Fragment>
+        {title}
+        <ArrowIcon name="keyboard_arrow_down" onClick={this.handleClick} />
+      </React.Fragment>
+    );
+  };
+
+  handleItemClick = onClick => {
```
Comment (ellis-cot) [inline src/components/DropdownButton/DropdownButton.tsx]: Don't think this is used anymore?

File: src/components/DropdownButton/DropdownButton.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    onClick();
+    this.setState({ isOpen: false });
+  };
+
+  render() {
+    const { isOpen } = this.state;
+    const { children, text, error, full, id, disabled, ...props } = this.props;
```
Comment (ellis-cot) [inline src/components/DropdownButton/DropdownButton.tsx]: I would rather we didn't spread the props going forwards. If you could remove the ...props from here and the {...props} from the OutsideAlerter please?

File: src/components/types.ts [hunk 3]
```diff
@@ -2,3 +2,8 @@ export interface Option {
   title: any;
   value: any;
 }
+
+export interface DropdownButtonItem {
```
Comment (ellis-cot) [inline src/components/types.ts]: not sure you need this anymore

File: src/components/atoms/Button/Button.tsx [hunk 4]
```diff
@@ -287,7 +287,7 @@ const Button: React.SFC<Props> = (props: Props) => {
         </LoadingWrapper>
       )}
       {!loading && children}
-      {icon && <Icon name={icon} />}
+      {icon && <Icon name={icon} {...rest} />}
```
Comment (ellis-cot) [inline src/components/atoms/Button/Button.tsx]: Seems a bit odd, the props are already spread in the parent

File: src/components/DropdownButton/DropdownButton.tsx [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+    /* Check if this is controlled by the parent, if yes, use the one from the parent */
+    // const selectedOption = isControllable ? this.props.selectedOption : this.state.selectedOption;
+    return (
+      <OutsideAlerter handleClickOutsideElement={this.handleClickOutsideDropdown} {...props}>
+        <Wrapper error={error} full={full} id={id} {...this.props} tabIndex={0}>
```
Comment (ellis-cot) [inline src/components/DropdownButton/DropdownButton.tsx]: again here, would be good if we don't {...this.props}

File: src/components/atoms/Button/Button.tsx (hunk 4 above)
Comment (AshleyKensington) [reply src/components/atoms/Button/Button.tsx]: But the icon didn't change colour with the text until I added this

File: src/components/types.ts (hunk 3 above)
Comment (AshleyKensington) [reply src/components/types.ts]: I had removed this.

File: src/components/DropdownButton/DropdownButton.tsx [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+const StyledButton = styled(Button)`
+  padding-right: 5px;
+  i {
+    float: right;
+    padding: 6px 0 0 6px;
+  }
+`;
+
+interface Props extends WrapperProps {
+  text?: any;
```
[the remark this answers is not in the document]
Comment (AshleyKensington) [reply src/components/DropdownButton/DropdownButton.tsx]: Changed to React.ReactNode

File: src/components/DropdownButton/DropdownButton.tsx (hunk 1 above)
Comment (AshleyKensington) [reply src/components/DropdownButton/DropdownButton.tsx]: Removed

File: src/components/DropdownButton/DropdownButton.tsx (hunk 2 above)
Comment (AshleyKensington) [reply src/components/DropdownButton/DropdownButton.tsx]: removed the props spreading

File: src/components/DropdownButton/DropdownButton.tsx (hunk 5 above)
Comment (AshleyKensington) [reply src/components/DropdownButton/DropdownButton.tsx]: removed

**Pull request merged**